src/interceptors/xhrInterceptor.js

```diff
@@ -72,7 +72,16 @@ const fakeService = helpers => class XMLHttpRequestInterceptor {
       //TODO: should 'this.response' be the response string or the response json?
       this.responseText = this.response = responseString;
       
-      return successCallback.call(this);
+      const requestDelay = helpers.getDelay();
+      if (!requestDelay) {
+        // for backwards compatibility if there is no requestDelay ot it 0 return callback without setTimeout
+        return successCallback.call(this);
+      } else {
+        return setTimeout(
+          () => successCallback.call(this),
+          requestDelay
+        );
+      }      
```

Comment:
That's a good point, let's keep like in your first version, it's safer as you said. Also could add a little test to check that the requestDelay works properly, you can check how we do in the `fetch` case [here](https://github.com/devlucky/Kakapo.js/blob/cb88c3605d94349f0341da8d3b15fea86fe916e2/test/specs/router_spec.js#L37-L60), notice the `assert.timeoutAfter(1100);` ;)